src/frontend/magnify/src/components/app/MagnifyProvider/index.tsx

```diff
@@ -41,26 +27,17 @@ const queryClient = new QueryClient({
 });
 
 export interface MagnifyProviderProps {
-  children?: React.ReactNode;
   initialUser?: User;
   translations?: any;
+  locale?: string;
 }
 
-const locale = 'en-US';
-
-export function MagnifyProvider({ ...props }: MagnifyProviderProps) {
-  const [translations, setTranslations] = useState<
-    Maybe<Record<string, string> | Record<string, MessageFormatElement[]>>
-  >({});
-
-  useEffect(() => {
-    getTranslation(locale).then((initTranslation) => {
-      setTranslations(initTranslation);
-    });
-  }, []);
-
+export function MagnifyProvider({
+  locale = 'en-US',
+  ...props
+}: PropsWithChildren<MagnifyProviderProps>) {
   return (
-    <TranslationProvider defaultLocale="en-US" locale={locale} messages={translations || {}}>
+    <TranslationProvider defaultLocale="en-US" locale={locale}>
```

Comment:
In case you do no want to make defaultLocale optional.
```suggestion
    <TranslationProvider defaultLocale={MagnifyLocales.EN} locale={locale}>
```

Reply:
He is optional now.